### ISSUE TITLE
Two definitions of transpose 

### ISSUE THREAD

**Issue by b-r-hamilton** [ggebbie/UnitfulLinearAlgebra.jl#54]
On the main branch, in src/base, there are two definitions of transpose (line 237 and 239)- I assume we want the second one? 

```
(237) Base.transpose(a::AbstractUnitfulVector) = rebuild(a,transpose(parent(a)),(Units([unit(1.0)]), Units(unitrange(a).^-1))) # kludge for unitrange of row vector
(238) Base.transpose(A::AbstractUnitfulDimMatrix) = rebuild(A,transpose(parent(A)),(Units(unitdomain(A).^-1), Units(unitrange(A).^-1)),(last(dims(A)),first(dims(A))))
(239) Base.transpose(a::AbstractUnitfulVector) = rebuild(a,transpose(parent(a)),(Units([unit(1.0)]), Units(unitrange(a).^-1)),(last(dims(a)),))
```

**Comment by ggebbie** [comment]
L239 should apply for AbstractUnitfulDimVector not AbstractUnitfulVector

then both L237 and L239 are needed

**Comment by ggebbie** [comment]
fixed in pull request #55
